### ISSUE TITLE
(WBS 1.14.66) Enable tabular preview for CSV files in the repository

### ISSUE THREAD

**Issue by ghjensen** [etcluvic/hsscommons#181]
Currently, the Repository allows visitors to preview publications that have PDFs, mp4s, or mp3s designated as the primary publication file. However, we would also like to enable similar previews for CSV files, too (and possibly even allow visitors to filter and/or search the CSVs). To be useful, the CSV previews should appear as HTML tables with the same width as the video previews (see, e.g., https://hsscommons.ca/en/publications/335/2#file-preview).

According to some preliminary searches I've conducted, two possible ways to implement this would involve using https://github.com/derekeder/csv-to-html-table (the best looking option, and my preference, if it's supported) **_OR_** the jQuery CSV to Table plugin (https://code.google.com/archive/p/jquerycsvtotable/). For reference, in case it's helpful, some information re: HUBzero's built-in Javascript support can also be found here: https://help.hubzero.org/documentation/22/webdevs/templates/javascript.

**Comment by kgosal03** [comment]
Hi @ghjensen , I have made changes and now the tables are visible as the way you wanted.

Two things, how long you want the tables to be in terms of height? Another thing is, do you want both xlsx and csv file types to be uploaded right? I mean they are same I guess just one is from excel and other is just a common format. Right now, it is supporting only csv file type. You can take a look at this link how the table  look as it is live on test server [link](https://test.hsscommons.ca/en/publications/2965/1). Also for tables, is there is some reason we want the width to be same as preview? For download CSV, if you directly click on it, it opens the link with some data, but if you right click and click save link as it downloads the file, I can take a look if I can change the behavior for this later but right now it works that way.

Also there is some issue going on with the test server since yesterday 4pm as I am not able to ssh into. It would be great if you can check on your side if you are able to ssh? I have emailed Matthew for this.

Thanks

**Comment by kgosal03** [comment]
@ghjensen I have made some changes to the table. I hope you like it. I tried to add the xlsx format but it is not working. So, we might have to stick with only csv format.

**Comment by ghjensen** [comment]
Hey @kgosal03 -- this looks amazing! The functionality (sorting, searching, pagination, download) is all really impressive. Great job. :)

Just a few small fine-tuning things I noticed:

1. Could we make the table full-width (like video previews in the repository), or at least much wider? We could also potentially shrink the table text and cell padding _slightly_, if necessary, to try make more data visible, but I think the width should be increased, if nothing else. Would that be possible?
2. You also asked about the height. I think the current height is fine. If it's possible to have the height increase or decrease based on the number you select in the "Show __ entries" drop-down, that would be cool, but I wouldn't worry about that if it's not easy to figure out.
3. Could we change the background color for the "Download as CSV" button from #5bc0de to #f1f1f1 (and then also change the text color from #fff to #777)? I like the blue you chose, but this revision would keep the button more consistent with existing buttons on the site.

Also, I noticed two strange (and seemingly unrelated) things:

1. The box with the "MANAGE THIS PUBLICATION" link (when you're viewing a publication you own) now has a grey background for some reason, instead of the original blue #6fb6ff background. Do you know what happened?
2. The font and text styling on the entire page seems to have changed... everything is now Arial, instead of using the usual font stack (font-family: 'OpenSans', 'Helvetica Neue', Helvetica, Arial, sans-serif;). Would it be possible to change back to the original font styling?

I know that looks like a long list—sorry. I really am excited about the new CSV display feature you've added!

**Comment by kgosal03** [comment]
Hi @ghjensen , thanks so much! I have addressed all the small things you mentioned.
1) Width is now full. I misunderstood earlier where I made the width as the pdf preview. Now, it is like Video preview.
2) As per height, now it shows by default 10 entries. If you change to 25 say, it will increase in height as desired.
3) I have changed the download button to be more consistent. Let me know if you like it. One minor thing I changed is gave the background color more darker version from the one you suggested as the button was not clearly visible. Again, if you don't like this I can change back.
4) The MANAGE PUBLICATION is back to the same color as the newly added plugins were messing with those.
5) Same with the font family so fixed that.
6) Also changed the functionality of the download CSV button as it was opening some weird csv content in the browser instead of downloading.

Let me know if anything else comes up otherwise  it is good to go. Thanks Graham :)
